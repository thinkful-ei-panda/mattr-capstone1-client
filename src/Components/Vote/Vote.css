
form.VoteForm {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}


.vote-input {
    width: 100%;
    margin: 10px;
    padding-left: 0px;
    padding: 5px;
    margin-left: auto;
    margin-right: auto;
    background: #03e9f4;
    margin-top: 10px;

}

@media screen and (min-width: 1200px) {
    form {
        width: 80%;

    }
    .vote-input {
        width: 100%;

    }
}