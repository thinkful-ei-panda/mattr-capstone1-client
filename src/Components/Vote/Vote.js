import React, { Component } from "react";
import ElectionApiService from "../../Services/election-api-service";
import history from "../../history";
import Democrat from "../Images/joe-biden.jpg";
import Republican from "../Images/donald-trump.jpeg";
import Nav from "../Nav";
import "./Vote.css";
export default class Vote extends Component {
  state = {
    hasError: false,
    election_id: 1,
    candidate_id: {
      value: null,
      touched: false,
    },
  };

  static getDerivedStateFromError(error) {
    return { hasError: true };
  }

  updateCandidateChoice(candidate_id) {
    this.setState({ candidate_id: { value: candidate_id, touched: true } });
  }

  handleVote = (event) => {
    event.preventDefault();
    const election_id = this.state.election_id;
    const candidate_id = this.state.candidate_id.value;

    ElectionApiService.postVote({
      election_id: election_id,
      candidate_id: candidate_id,
    })
      .then((res) => {
      candidate_id.value = "";

        if(!res.ok){
          return res.json().then(e => Promise.reject(e))
        }

        return res.json();
      })
      .then((res) => {
        history.push("/VoteConfirmation");
      })
      .catch((error) => {
        history.push("/VoteError");
      });
  };

  render() {
    return (
      <div>
        <Nav />
        <form className="VoteForm" onSubmit={this.handleVote}>
          <h2> Presidential Election 2020 </h2>
          <div className="election-cards">
            <div className="scorecard-entry">
              <img src={Democrat} alt="Joe Biden" />
              <div className="stats">
                <input
                  type="radio"
                  value="1"
                  name="presidential-election-2020"
                  id="candidate_id"
                  onChange={(e) => this.updateCandidateChoice(e.target.value)}
                />
                <label htmlFor="joe-biden-vote"> Joe Biden </label>
              </div>
            </div>

            <div className="scorecard-entry">
              <img src={Republican} alt="Donald Trump" />
              <div className="stats">
                <input
                  type="radio"
                  value="2"
                  name="presidential-election-2020"
                  id="candidate_id"
                  onChange={(e) => this.updateCandidateChoice(e.target.value)}
                />
                <label htmlFor="donald-trump-vote">Donald Trump</label>
              </div>
            </div>
          </div>

          <button type="submit" className="vote-link">
            <span></span>
            <span></span>
            <span></span>
            <span></span>
            Cast Vote
          </button>
        </form>
      </div>
    );
  }
}
