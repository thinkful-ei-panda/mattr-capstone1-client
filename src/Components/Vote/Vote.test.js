import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Vote from "./Vote";

describe(`Vote Component`, () => {
  describe(`Smoke test`, () => {
    it(`Renders without crashing`, () => {
      const div = document.createElement("div");
      ReactDOM.render(
        <BrowserRouter>
          <Vote />
        </BrowserRouter>,
        div
      );
      ReactDOM.unmountComponentAtNode(div);
    });
  });

  describe(`Snapshot test`, () => {
    it(`Renders the UI as expected`, () => {
      const tree = renderer
        .create(
          <BrowserRouter>
            <Vote />
          </BrowserRouter>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
