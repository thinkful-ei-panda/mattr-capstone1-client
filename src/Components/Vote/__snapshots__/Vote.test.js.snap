// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vote Component Snapshot test Renders the UI as expected 1`] = `
<div>
  <div
    className="Nav"
  >
    <a
      href="/Home"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/Election"
      onClick={[Function]}
    >
      Elections
    </a>
    <a
      href="/RegistrationForm"
      onClick={[Function]}
    >
       
      Register
       
    </a>
    <a
      href="/Login"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
  <form
    className="VoteForm"
    onSubmit={[Function]}
  >
    <h2>
       Presidential Election 2020 
    </h2>
    <div
      className="election-cards"
    >
      <div
        className="scorecard-entry"
      >
        <img
          alt="Joe Biden"
          src="joe-biden.jpg"
        />
        <div
          className="stats"
        >
          <input
            id="candidate_id"
            name="presidential-election-2020"
            onChange={[Function]}
            type="radio"
            value="1"
          />
          <label
            htmlFor="joe-biden-vote"
          >
             Joe Biden 
          </label>
        </div>
      </div>
      <div
        className="scorecard-entry"
      >
        <img
          alt="Donald Trump"
          src="donald-trump.jpeg"
        />
        <div
          className="stats"
        >
          <input
            id="candidate_id"
            name="presidential-election-2020"
            onChange={[Function]}
            type="radio"
            value="2"
          />
          <label
            htmlFor="donald-trump-vote"
          >
            Donald Trump
          </label>
        </div>
      </div>
    </div>
    <button
      className="vote-link"
      type="submit"
    >
      Cast Vote
    </button>
  </form>
</div>
`;
